lib/rules/at-rule-no-deprecated/index.mjs

```diff
@@ -0,0 +1,65 @@
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { deprecatedAtKeywords } from '../../reference/atKeywords.mjs';
+import isStandardSyntaxAtRule from '../../utils/isStandardSyntaxAtRule.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'at-rule-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (atRule) => `Unexpected deprecated at-rule "${atRule}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-no-deprecated',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreAtRules: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkAtRules((atRule) => {
+			if (!isStandardSyntaxAtRule(atRule)) return;
+
+			const { name } = atRule;
+
+			if (optionsMatches(secondaryOptions, 'ignoreAtRules', name)) return;
+
+			if (!deprecatedAtKeywords.has(name.toLowerCase())) return;
+
+			const atName = `@${name}`;
+
+			report({
+				message: messages.rejected,
+				messageArgs: [atName],
+				node: atRule,
+				ruleName,
+				result,
+				index: 0,
+				endIndex: atName.length,
+			});
```

Comment:
```suggestion
				endIndex: atName.length,
				fix: name.toLowerCase() === 'nest' ? () => {
					// implies `import postcss from 'postcss';` at the top
					const styleRule = postcss.rule({
						selector: atRule.params,
						source: atRule.source,
					});

					if (atRule.nodes) {
						atRule.nodes.forEach((node) => {
							styleRule.append(node);
						});
					}

					atRule.replaceWith(styleRule);
				} : undefined
			});
```

[suggestion] `@nest` can be auto fixed by changing the node to a regular rule.

See: https://github.com/csstools/postcss-plugins/blob/main/plugins-stylelint/no-at-nest-rule/index.mjs

_The suggested code is not in the ideal form or entirely following our conventions. It is only intended as an illustration._

Reply:
> I figured I'd quickly create this one…

If we want to expedite this PR should we separate the fixer from the addition of the rule itself?